apps/src/code-studio/components/exportDialogRedux.js

```diff
@@ -2,25 +2,35 @@
 
 const SHOW_EXPORT_DIALOG = 'exportDialog/SHOW_EXPORT_DIALOG';
 const HIDE_EXPORT_DIALOG = 'exportDialog/HIDE_EXPORT_DIALOG';
+const SET_EXPORT_GENERATED_PROPERTIES =
+  'exportDialog/SET_EXPORT_GENERATED_PROPERTIES';
 
 // Reducer
 
 const initialState = {
-  isOpen: false
+  isOpen: false,
+  exportGeneratedProperties: {}
 };
 
 export default function reducer(state = initialState, action) {
   switch (action.type) {
     case SHOW_EXPORT_DIALOG:
       return {
-        ...state,
         ...initialState,
+        ...state,
```

Comment:
Rather than doing this for each case, move this to the top of the function. i.e. line 16:
`state = {...initialState, ...state}`

Reply:
yes, this copied a pattern from `shareDialogRedux.js`, but there is actually a bug there (`initialState` is spread after `state`). I think since the function already has `(state = initialState)`, I'm just going to remove `...initialState` from each location.

Reply:
Not worried if isOpen or exportGeneratedProperties aren't set?

Reply:
These two variables are set. When redux first calls the reducer, the first argument (`state`) is `undefined`, so the default param assignment (`reducer(state = initialState, action)`) ensures that state starts off with the `initialState` values (including `isOpen` and `exportGeneratedProperties`). They are returned back as part of the `state` for future operations.

I verified this behavior in the debugger.